tests/test_mysql_binlog_json.py

```diff
@@ -0,0 +1,333 @@
+        props = self.get_properties()
+        creds = self.get_credentials()
+
+        connection = pymysql.connect(host=props['host'],
+                                     database=props['database'],
+                                     port=int(props['port']),
+                                     user=props['user'],
+                                     password=creds['password'],
+                                     autocommit=True)
+
+        with connection.cursor() as cur:
+            cur.execute('SET @@session.time_zone="+0:00"')
+        return connection
+
+
+    def table_name(self):
+        return "mysql_binlog_json_test"
+
+
+    def tap_stream_id(self):
+        return "{}-{}".format(self.database_name(), self.table_name())
+
+
+    def insert_record(self, cursor, record):
+        rec_cols, rec_vals = zip(*record.items())
+
+        columns_sql = ", \n".join(rec_cols)
+        value_sql = ",".join(["%s" for i in range(len(rec_vals))])
+
+        insert_sql = """
+        INSERT INTO {}.{}
+               ( {} )
+        VALUES ( {} )""".format(
+            self.database_name(),
+            self.table_name(),
+            columns_sql,
+            value_sql)
+
+        cursor.execute(insert_sql, rec_vals)    
+
+
+    def setUp(self):
+        missing_envs = [x for x in [os.getenv('TAP_MYSQL_HOST'),
+                                    os.getenv('TAP_MYSQL_PORT'),
+                                    os.getenv('TAP_MYSQL_USER'),
+                                    os.getenv('TAP_MYSQL_PASSWORD'),
+                                    os.getenv('TAP_MYSQL_PORT')] if x == None]
+        if len(missing_envs) != 0:
+            #pylint: disable=line-too-long
+            raise Exception("set TAP_MYSQL_HOST, TAP_MYSQL_PORT, TAP_MYSQL_DBNAME, TAP_MYSQL_USER, TAP_MYSQL_PASSWORD")
+
+        connection = self.get_db_connection()
+
+        with connection.cursor() as cur:
+            create_databases_sql = """
+                DROP DATABASE IF EXISTS {};
+                CREATE DATABASE {};
+            """.format(self.database_name(), self.database_name())
+
+            cur.execute(create_databases_sql)
+
+            cur.execute("""
+            SELECT EXISTS (
+            SELECT 1
+            FROM  information_schema.tables
+            WHERE  table_schema = %s
+            AND  table_name =   %s);""",
+                        [self.database_name(), self.table_name()])
+
+            existing_table = cur.fetchone()[0]
+
+            if existing_table:
+                cur.execute("DROP TABLE {}.{}".format(self.database_name(), self.table_name()))
+
+            create_table_sql = """
+CREATE TABLE {}.{} (
+            id                     BIGINT PRIMARY KEY,
+            our_json               JSON
+)
+""".format(self.database_name(), self.table_name())
+
+            cur.execute(create_table_sql)
+
+            for record in [rec_1]:
+                self.insert_record(cur, record)
+
+
+    def test_run(self):
```

Comment:
Again, comparing to the other file, there's definitely repeated code between the two. I think it would both be easier to read (by reducing cognitive load) and DRY up the test if there was a base method where some repeated code could be placed.

Reply:
Mhm, these are old tests. I could put an issue on the repo maybe for cleaning up?

Reply:
Maybe instead of an issue about test cleanup for this specific test we write an issue for some common test methods to add to tap-tester that nearly 100% of our tests use?